tests/layer_tests/onnx_tests/test_loop.py

```diff
@@ -0,0 +1,401 @@
+            outputs=['cond_out_2'],
+        )
+
+        in_2_int_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['in_2_int'],
+            outputs=['in_2_int_out'],
+        )
+
+        body_graph_2 = helper.make_graph(
+            [one_in_2, add_node, cond_2_node, in_2_int_node, one_int_in_2, add_one_to_m_2_node],
+            'body_graph_2',
+            [m_2, cond_int_2, in_2_int],
+            [cond_out_2, in_2_int_out, out_2],
+        )
+
+        node_loop_2 = onnx.helper.make_node(
+            'Loop',
+            inputs=['M_2', 'cond_2', 'IN_2'],
+            outputs=['cond_out_2', 'OUT_2'],
+            body=body_graph_2
+        )
+
+        out_1_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['OUT_2'],
+            outputs=['OUT_1'],
+        )
+
+        cond_1_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['cond_int_1'],
+            outputs=['cond_out_1'],
+        )
+
+        in_1_int_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['in_1_int'],
+            outputs=['in_1_int_out'],
+        )
+
+        body_graph_1 = helper.make_graph(
+            [one, add_one_node, one_int, add_one_to_m_node, M_2, cond_2, node_loop_2, out_1_node, cond_1_node,
+             in_1_int_node],
+            'body_graph_1',
+            [m_1, cond_int_1, in_1_int],
+            [cond_out_1, in_1_int_out, out_1],
+        )
+
+        node_loop_1 = onnx.helper.make_node(
+            'Loop',
+            inputs=['M_1', 'cond', 'IN_1'],
+            outputs=['cond_out_1', 'OUT_1'],
+            body=body_graph_1
+        )
+
+        res_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['OUT_1'],
+            outputs=['res'],
+        )
+
+        graph_def = helper.make_graph(
+            [M_1, cond, node_loop_1, res_node],
+            'graph',
+            [in_1, in_2],
+            [res],
+        )
+
+        onnx_net = helper.make_model(graph_def, producer_name='test_loop_in_loop_model')
+        # We do not create reference graph, as it's too complicated to construct it
+        # So we return None to skip IR comparision
+
+        return onnx_net, None
+
+    @pytest.mark.precommit
+    @pytest.mark.timeout(250)
+    def test_loop_simple_precommit(self, ie_device, precision, ir_version, temp_dir):
+        if ie_device == 'GPU':
+            pytest.skip('Loop not supported on GPU')
+        self._test(*self.create_loop(), ie_device, precision, ir_version, temp_dir=temp_dir,
+                   infer_timeout=150)
+
+    @pytest.mark.precommit
+    @pytest.mark.timeout(250)
+    def test_loop_in_loop_simple_precommit(self, ie_device, precision, ir_version, temp_dir):
+        if ie_device == 'GPU':
+            pytest.skip('Loop not supported on GPU')
+        self._test(*self.create_loop_in_loop(), ie_device, precision, ir_version, temp_dir=temp_dir,
+                   infer_timeout=150)
```

Comment:
do we cover a case with transit parameter's node?

Reply:
yes